skimage/future/graph/rag.py

```diff
@@ -321,6 +382,100 @@ def rag_mean_color(image, labels, connectivity=2, mode='distance',
     return graph
 
 
+def rag_generic(image, labels, connectivity=2, describe_func=describe_node,
+                weight_func=mean_color, extra_arguments=[],
+                extra_keywords={'sigma':255.0, 'mode':'distance'}):
```

Comment:
A few comments on this signature:

0. Maybe rename the function, simply, `region_adjacency_graph`?
1. `labels` should go before `image`, because it is possible to create a RAG without an image, but not without a label map. Indeed, `image` can default to `None`.
2. I don't like `connectivity=2` as the default because it makes little sense for 3D images. `connectivity=1` works, as well as `connectivity=None` which would get replaced by `labels.ndim` inside the function. I have no strong preference between these two options, but I *slightly* prefer `1` because it can be read in the function signature, rather than the docstring.
3. I think the callbacks should default to `None`. A basic RAG creation function should just create a RAG, based on the definition (adjacent labels have an edge in the graph), and nothing more.
4. In addition to node descriptions, there will also, in many cases, be edge descriptions. So we need both `node_describe_func` (maybe `node_descriptor`?) and `edge_describe_func`.
5. (I don't have a solution for this right now, but something to think about.) There is a very strong coupling between the `describe_func` and `weight_func`. Indeed, it would almost certainly be a 1-1 mapping between descriptors and merge functions. Given this, it might be worthwhile thinking about how to encapsulate these two parameters into a single one.

These comments are just details, though. Overall I have to concur with @JDWarner: a very clean first PR! =)